crypto/src/secp256k1.rs

```diff
@@ -279,8 +280,8 @@ impl KeyPair for Secp256k1KeyPair {
         self.secret
     }
 
-    fn generate<R: rand::CryptoRng + rand::RngCore>(rng: &mut R) -> Self {
-        let privkey = k256::ecdsa::SigningKey::random(rng);
+    fn generate<R: rand::CryptoRng + rand::RngCore>(_rng: &mut R) -> Self {
+        let privkey = k256::ecdsa::SigningKey::random(&mut OsRng);
```

Comment:
workaround here: instead of using what R is passed in (the version for rand is different from what k256 library likes to take in), generate a fresh OsRng that complies to the k256 interface

Reply:
Yep. Please add a TODO to:
1. clean this hack up once we upgrade our eddsa implementation,
2. NOT use this in tests: we have several fixtures that generate / copy the same keys in tests repeatedly by passing in carefully-seeded RNGs (see `keys()` in `test_utils`).